core/src/main/scala/chisel3/VerificationStatement.scala

```diff
@@ -432,23 +432,33 @@ private object VerificationStatement {
     (p.source.file.name, p.line, p.lineContent.trim): @nowarn // suppress, there's no clear replacement
   }
 
-  def failureMessage(
+  def formatFailureMessage(
     kind:     String,
     lineInfo: SourceLineInfo,
     cond:     Bool,
     message:  Option[Printable]
   )(
     implicit sourceInfo: SourceInfo
-  ): Unit = {
+  ): Printable = {
     val (filename, line, content) = lineInfo
     val lineMsg = s"$filename:$line $content".replaceAll("%", "%%")
-    val fmt = message match {
+    message match {
       case Some(msg) =>
         p"$kind failed: $msg\n    at $lineMsg\n"
       case None => p"$kind failed\n    at $lineMsg\n"
     }
```

Comment:
Incidentally I'm about to PR getting rid of lineMsg, let's get your PR in first.

Reply:
Great, and thanks for letting this go first, sorry for the delay :+1: .